dashboard/api_edx_cache.py

```diff
@@ -16,42 +16,31 @@
 
 
 class CachedEdxUserData:
-    """Represents all edx data related to a User"""
-    # pylint: disable=too-many-instance-attributes
-
-    enrollments = None
-    certificates = None
-    current_grades = None
-    raw_enrollments = None
-    raw_certificates = None
-    raw_current_grades = None
+    """Represents all edX data related to a User"""
+    edx_data_tuple = namedtuple('EdxDataTuple', ['queryset', 'edx_object'])
 
-    def __init__(self, user, program=None, include_raw_data=False):
-        """
-        Fetches the given User's edx data and sets object properties
-
-        Args:
-            user (User): a User object
-            program (Program): an optional Program to filter on
-            include_raw_data (bool): Set to True if raw edx data is needed (in addition to actual edX objects)
-        """
+    def __init__(self, user, program=None):
+        """Instantiates object and fetches all of a user's cached edX data"""
         self.user = user
         self.program = program
-        self.fetch_and_set_edx_data()
-        if include_raw_data:
-            self.fetch_and_set_raw_data()
-
-    def fetch_and_set_edx_data(self):
-        """Fetches edx data and sets object properties"""
-        self.enrollments = models.CachedEnrollment.get_edx_data(self.user, program=self.program)
-        self.certificates = models.CachedCertificate.get_edx_data(self.user, program=self.program)
-        self.current_grades = models.CachedCurrentGrade.get_edx_data(self.user, program=self.program)
-
-    def fetch_and_set_raw_data(self):
-        """Fetches raw cached data and sets object properties"""
-        self.raw_enrollments = list(models.CachedEnrollment.data_qset(self.user, program=self.program))
-        self.raw_certificates = list(models.CachedCertificate.data_qset(self.user, program=self.program))
-        self.raw_current_grades = list(models.CachedCurrentGrade.data_qset(self.user, program=self.program))
+        self.enrollments = self.fetch_and_set_edx_data_tuple(models.CachedEnrollment)
+        self.certificates = self.fetch_and_set_edx_data_tuple(models.CachedCertificate)
+        self.current_grades = self.fetch_and_set_edx_data_tuple(models.CachedCurrentGrade)
+
+    def fetch_and_set_edx_data_tuple(self, model_cls):
+        """Fetches a user's edX data, deserializes it to an edX API object, and returns a named tuple"""
+        queryset = model_cls.data_qset(self.user, program=self.program)
+        edx_object = model_cls.deserialize_edx_data(queryset)
+        return self.edx_data_tuple(queryset, edx_object)
```

Comment:
IMHO this is getting over complicated because too optimized. I think you should have two methods to get the two pieces of data and until we realize that that is a performance problem we should not worry about that.